### DISCUSSION TITLE
Working with Docker image deps inside pom.xml files

### DISCUSSION THREAD

**Discussion by hoshposh** [renovatebot/renovate#26732] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.137.2

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### What is your question?

I have maven projects that build Docker images. The `Dockerfile` uses `ARG` declaration so that the image tags for the `FROM` declarations can be passed in from the `pom.xml`. What are the correct approaches so that the Docker dependencies are updated in the Maven build file?

I have tried searching this discussion area and external websites and have found approaches that rely on [regex manager](https://github.com/renovatebot/renovate/discussions/11104#discussioncomment-1133482). Is that the approach needed here?

**Dockerfile**:

```Dockerfile
ARG BASE_IMAGE

FROM ${BASE_IMAGE} AS builder
```

**pom.xml**:

```xml
  <properties>
    <internal.base.image>internal-path/internal-base-image:5.13.0</internal.base.image>
  </properties>
  .
  .
      <plugin>
        <groupId>io.fabric8</groupId>
        <artifactId>docker-maven-plugin</artifactId>
        <executions>
          <execution>
            <id>artifactory-docker-build</id>
            <phase>compile</phase>
            <goals>
              <goal>build</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <images>
            <image>
              <build>
                <dockerFileDir>.</dockerFileDir>
                <args>
                  <BASE_IMAGE>${internal.base.image}</BASE_IMAGE>
                </args>
              </build>
            </image>
          </images>
        </configuration>
      </plugin>
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Define your own customManagers

**Reply by tdharris** [reply]
Example `pom.xml` :
```xml
<properties>
    <!-- renovate auto-update-github-releases: -->
    <internal.base.image>internal-path/internal-base-image:5.13.0</internal.base.image>
</properties>
```

With a [customManager](https://docs.renovatebot.com/configuration-options/#custommanagers) like the following:

```json
{
    "fileMatch": ["\\pom.xml$"],
    "matchStrings": ["renovate auto-update-(?<datasource>.+?):.*\n\s+<.+\.base\.image>(?<depName>.+?):(?<currentValue>.+?)<\/.+\.base\.image>"]
},
```

Would extract:

  - datasource: `github-releases`
  - depName: `internal-path/internal-base-image`
  - currentValue: `5.13.0`

**Reply by hoshposh** [reply, marked as answer]
I elected to follow your advice and use this `customManagers` definition:

```json
  "customManagers": [
    {
      "customType": "regex",
      "datasourceTemplate": "docker",
      "fileMatch": [
        "(^|/)pom\\.xml$"
      ],
      "matchStrings": [
        "<internal\\.base\\.image>(?<depName>[^\\s]+?):(?<currentValue>.*)<\\/internal\\.base\\.image>"
      ],
      "versioningTemplate": "docker"
    }
  ],
```
